### PR TITLE
(DOCSP-21862): Bundle Realm for React Native Rehash 

### PR DESCRIPTION
**Pull request by bpmutter**
## Pull Request Info

refactor the React Native create a bundled realm page b.c the flow is fairly separate from the Node.Js page. this change corrects multiple errors in the current page. 

to accomplish this change, i had to create a new RN/realm app with in the node examples. 

i also provided some small updates to the expo app as part of the PR. 

### Jira

- https://jira.mongodb.org/browse/DOCSP-21862

### Staged Changes (Requires MongoDB Corp SSO)

- [Bundle a Realm (React Native)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-21862/sdk/react-native/advanced/bundle/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/advanced/bundle.txt [hunk 1]
```diff
@@ -22,106 +22,111 @@ initial download the first time a user opens your application. Instead,
...
-.. code-block::
-   :caption: prod_realm_app
-   :emphasize-lines: 2
+         .. tab:: iOS
+            :tabid: ios
 
-   .
-   ├── copyOfDefault.realm 
-   ... rest of files in _prod_ application
+            #. Open up the ``ios`` directory in Xcode.
```
Comment (tomduncalf) [inline source/sdk/react-native/advanced/bundle.txt:102]: You actually need to open the `xcworkspace`. Maybe there's a better way to word it than this but:

```suggestion
            #. Open up the ``your_project_name.xcworkspace`` file (in the ``ios`` directory generated by React Native)  in Xcode.
```

File: source/sdk/react-native/advanced/bundle.txt [hunk 2]
```diff
@@ -22,106 +22,111 @@ initial download the first time a user opens your application. Instead,
...
+            #. Select you app in the project navigator and
```
Comment (tomduncalf) [inline source/sdk/react-native/advanced/bundle.txt:103]:
```suggestion
            #. Select your app in the project navigator and
```

File: source/sdk/react-native/advanced/bundle.txt [hunk 3]
```diff
@@ -22,106 +22,111 @@ initial download the first time a user opens your application. Instead,
...
+               select the :guilabel:`Build Phases` tab in the project overview.
+            #. Select the :guilabel:`Copy Bundle Resources` menu. Click the :guilabel:`+` icon.
```
Comment (tomduncalf) [inline source/sdk/react-native/advanced/bundle.txt:105]:
```suggestion
            #. Expand the :guilabel:`Copy Bundle Resources` item.
            #. Click the :guilabel:`+` icon. 
```

File: source/sdk/react-native/advanced/bundle.txt [hunk 4]
```diff
@@ -22,106 +22,111 @@ initial download the first time a user opens your application. Instead,
...
 
-   .
-   ├── copyOfDefault.realm 
-   ... rest of files in _prod_ application
+            #. Open up the ``ios`` directory in Xcode.
+            #. Select you app in the project navigator and
+               select the :guilabel:`Build Phases` tab in the project overview.
+            #. Select the :guilabel:`Copy Bundle Resources` menu. Click the :guilabel:`+` icon. 
+            #. Find your bundled realm file and add it. Don't change the default settings,
+               then select the :guilabel`Finish` button.
```
Comment (tomduncalf) [inline source/sdk/react-native/advanced/bundle.txt:107]:
```suggestion
               then select the :guilabel:`Finish` button.
```

**Review by tomduncalf — CHANGES_REQUESTED** [review]
LGTM aside from requested changes

File: source/sdk/react-native/advanced/bundle.txt [hunk 5]
```diff
@@ -22,106 +22,113 @@ initial download the first time a user opens your application. Instead,
...
-#. :ref:`Add the bundled {+realm+} file <react-native-bundle-realm-file>` 
-   to your production application.
+Follow these steps to create and bundle a {+realm+} file for your React Native application.
+On a high level you:
 
-#. In your production application,
-   :ref:`open the {+realm+} from the bundled asset file
-   <react-native-open-a-realm-from-a-bundled-realm-file>`. For synchronized
-   {+realm+}s, you must include the partition key.
+#. Create a bundled realm file in a Node.Js script.
```
Comment (nlarew) [inline source/sdk/react-native/advanced/bundle.txt:36]:
```suggestion
#. Create a bundled realm file in a Node.js script.
```

File: source/sdk/react-native/advanced/bundle.txt [hunk 6]
```diff
@@ -22,106 +22,113 @@ initial download the first time a user opens your application. Instead,
...
 
 .. _react-native-create-a-realm-for-bundling:
 
-Create a Realm File for Bundling
---------------------------------
+.. procedure:: 
 
-1. Build a temporary {+realm+} app that shares the data model of your
-   application.
+   .. step:: Create Realm File for Bundling
```
Comment (nlarew) [inline source/sdk/react-native/advanced/bundle.txt:53]:
```suggestion
   .. step:: Create a Realm File to Bundle
```

File: source/sdk/react-native/advanced/bundle.txt [hunk 7]
```diff
@@ -22,106 +22,113 @@ initial download the first time a user opens your application. Instead,
...
 
-.. _react-native-bundle-realm-file:
+      Now that you have a copy of the {+realm+} that contains the initial data,
+      add the bundled {+realm+} file to your production application.
+      Where you place the bundled realm differs for iOS and Android 
+      builds. 
 
-Bundle a Realm File in Your Production Application
-----------------------------------------------------
+      .. tabs::
```
Comment (nlarew) [inline source/sdk/react-native/advanced/bundle.txt:89]: Is there a shell one-liner we can give folks for each platform? I don't love how this is tied to specific UI instructions (though it's probably fine)

File: source/sdk/react-native/advanced/bundle.txt [hunk 8]
```diff
@@ -22,106 +22,113 @@ initial download the first time a user opens your application. Instead,
...
 
-   .. code-block::
-      :caption: temp_realm_app
-      :emphasize-lines: 2
+      .. _react-native-bundle-realm-file:
 
-      .
-      ├── copyOfDefault.realm
-      ... rest of files in _temp_ application
+   .. step:: Bundle Realm File in Your Production Application
```
Comment (nlarew) [inline source/sdk/react-native/advanced/bundle.txt:82]:
```suggestion
   .. step:: Bundle the Realm File with your App
```
nit: doesn't need to be a prod app necessarily

File: source/sdk/react-native/advanced/bundle.txt [hunk 9]
```diff
@@ -22,106 +22,113 @@ initial download the first time a user opens your application. Instead,
...
+            #. Click the :guilabel:`Add Other...` button.
+            #. Find your bundled realm file and add it. Don't change the default settings,
+               then select the :guilabel:`Finish` button.
 
-.. _react-native-open-a-realm-from-a-bundled-realm-file:
+      .. _react-native-open-a-realm-from-a-bundled-realm-file:
 
-Open a Realm from a Bundled Realm File
---------------------------------------
+   .. step:: Open a Realm from Bundle
```
Comment (nlarew) [inline source/sdk/react-native/advanced/bundle.txt:113]:
```suggestion
   .. step:: Open the Bundled Realm in your App
```

File: source/sdk/react-native/advanced/bundle.txt [hunk 10]
```diff
@@ -22,106 +22,113 @@ initial download the first time a user opens your application. Instead,
...
 
-Now that you have a copy of the {+realm+} included with your production
-application, you need to add code to use it. 
+      To add the bundled realm file to your app's document directory, call 
+      :js-sdk:`Realm.copyBundledRealmFiles() <Realm.html#.copyBundledRealmFiles>`
+      before you :ref:`open the realm <react-native-open-close-realm>`.
```
Comment (nlarew) [inline source/sdk/react-native/advanced/bundle.txt:117]: Could we add a little context at the start of this step? e.g.

- The realm is now bundled and will be included whenever a user downloads the app
- To use the bundled data, you have to copy it into a new realm file in your app's document directory
- To add the bundled realm file to your app's document directory....

**Review by nlarew — APPROVED** [review]
A few suggestions but overall LGTM

File: source/sdk/react-native/advanced/bundle.txt (hunk 7 above)
Comment (mongodben) [reply source/sdk/react-native/advanced/bundle.txt:89]: dont think so unfortunately :(

File: source/sdk/react-native/advanced/bundle.txt (hunk 10 above)
Comment (mongodben) [reply source/sdk/react-native/advanced/bundle.txt:117]: adding suggestion 1

**Pull request merged**